hal/network/ncp/at_parser/at_parser.h

```diff
@@ -446,4 +470,13 @@ inline bool AtParserConfig::logEnabled() const {
     return logEnabled_;
 }
 
+inline AtParserConfig& AtParserConfig::logCategory(const char* category) {
+    logCategory_ = category;
+    return *this;
+}
+
+inline const char* AtParserConfig::logCategory() const {
+    return logCategory_ ? static_cast<const char*>(logCategory_) : DEFAULT_LOG_CATEGORY;
```

Comment:
Why not use the parenthesis operator?

Reply:
`CString` is not a functor, it doesn't have an `operator()` it has a cast operator to `const char*`: https://en.cppreference.com/w/cpp/language/cast_operator. Ternary if also requires both variants to be of the same type, so an explicit case is required here, otherwise the compiler will throw an error.